jupyter/d2l-java/chapter_convolutional-neural-networks/lenet.ipynb

```diff
@@ -0,0 +1,577 @@
+    "    // to see if we have multiple samples\n",
+    "    if (yHat.getShape().size(1) > 1) {\n",
+    "        // Argmax gets index of maximum args for given axis 1\n",
+    "        // Convert yHat to same dataType as y (int32)\n",
+    "        // Sum up number of true entries\n",
+    "        return yHat.argMax(1).toType(DataType.INT32, false).eq(y.toType(DataType.INT32, false))\n",
+    "                .sum().toType(DataType.FLOAT32, false).getFloat();\n",
+    "    }\n",
+    "    return yHat.toType(DataType.INT32, false).eq(y.toType(DataType.INT32, false))\n",
+    "            .sum().toType(DataType.FLOAT32, false).getFloat();\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The training function `trainChapter6` is also similar \n",
+    "to `trainChapter3` defined in :numref:`sec_softmax_scratch`. \n",
+    "Since we will be implementing networks with many layers \n",
+    "going forward, we will rely primarily on DJL.\n",
+    "The following train function assumes a Gluon model \n",
+    "as input and is optimized accordingly. \n",
+    "We initialize the model parameters \n",
+    "on the block using the Xavier initializer.\n",
+    "Just as with MLPs, our loss function is cross-entropy,\n",
+    "and we minimize it via minibatch stochastic gradient descent. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "int numEpochs = 10;\n",
+    "float lr = 0.9f;\n",
+    "\n",
+    "double[] trainLoss;\n",
+    "double[] testAccuracy;\n",
+    "double[] epochCount;\n",
+    "double[] trainAccuracy;\n",
+    "\n",
+    "trainLoss = new double[numEpochs];\n",
+    "trainAccuracy = new double[numEpochs];\n",
+    "testAccuracy = new double[numEpochs];\n",
+    "epochCount = new double[numEpochs];"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "public void trainChapter6(){\n",
+    "    \n",
+    "    Loss loss = Loss.softmaxCrossEntropyLoss();\n",
+    "\n",
+    "    LearningRateTracker lrt = LearningRateTracker.fixedLearningRate(lr);\n",
+    "    Optimizer sgd = Optimizer.sgd().setLearningRateTracker(lrt).setRescaleGrad(1.0f/batchSize).build();\n",
+    "\n",
+    "    DefaultTrainingConfig config = new DefaultTrainingConfig(loss)\n",
+    "                    .optOptimizer(sgd) // Optimizer (loss function)\n",
+    "                    .addEvaluator(new Accuracy()) // Model Accuracy\n",
+    "                    .addTrainingListeners(TrainingListener.Defaults.logging()); // Logging\n",
+    "\n",
+    "    Model model = Model.newInstance(Device.defaultDevice().toString());\n",
+    "    model.setBlock(block);\n",
+    "    Trainer trainer = model.newTrainer(config);\n",
+    "    \n",
+    "    float epochLoss = 0f;\n",
+    "    float accuracyVal = 0f;\n",
+    "    long totalTime = 0;\n",
```

Comment:
I'm thinking it might be better to use EasyTrain.fit here. I think we are past the point of training from scratch at this stage in the book and we should focus on practical training and the model designs. You could also use the DJL Accuracy instead of reimplementing it above

Reply:
So the only reason I went for the normal scratch method was to get the examples/sec on gpu stats. If that isn't required, I was initially moving to the EasyTrain method only. 
If we can get that stat in some other way, I can do that also.

Reply:
We record the time per batch to a metric in the TimeMeasureTrainingListener. You can use metrics.mean and divide by batch size to get the time per example. This information might also be printed by the logging training listener